examples/ping_client/src/main.rs

```diff
@@ -80,7 +80,13 @@ mod ping_client {
             //   for no more than the refresh_period
             // - `bytes` is a subslice of `buf`, containing only the bytes actually read
             //   from the sampling port
-            let (validity, bytes) = ctx.ping_response.unwrap().receive(&mut buf).unwrap();
+            let (validity, bytes) = match ctx.ping_response.unwrap().receive(&mut buf) {
+                Ok((validity, bytes)) => (validity, bytes),
+                Err(e) => {
+                    warn!("Failed to receive ping response: {e:?}");
+                    continue;
+                }
+            };
```

Comment:
Maybe we could use `let-else` https://doc.rust-lang.org/beta/rust-by-example/flow_control/let_else.html because a `continue` should qualify as a diverge

Reply:
We just had this discussion, its the same. If you use continue, or diverge in a pattern match doesn't make a difference.